src/streamlit_extras/prometheus/__init__.py

```diff
@@ -0,0 +1,139 @@
+from typing import List, NamedTuple
+
+from prometheus_client import CollectorRegistry
+from prometheus_client.openmetrics.exposition import generate_latest
+from streamlit.runtime.stats import CacheStatsProvider
```

Comment:
Is this import something that's been valid for a few releases of streamlit already? We don't pin streamlit to anything else then > 1.0.0 so if this was introduced only in 1.31.0, that may break for others. LMK

Reply:
(I read `**Note:** This extra works best with Streamlit >= 1.31. There are known bugs with
    some earlier Streamlit versions, especially 1.30.0.`, just curious if it would bug or break with prior)

Reply:
Yes, this import has [worked since 1.12](https://github.com/streamlit/streamlit/blob/1.12.0/lib/streamlit/runtime/stats.py).

There was a bug in Streamlit's native implementation of Prometheus since before 1.12 -> 1.30 where it printed duplicate metric rows that breaks the protocol format. I think in practice most collectors would be able to handle it but it might throw a warning or something (and we never surfaced the endpoint for real use so no one complained). So this collector would work prior to 1.30 but the Streamlit portion might be slightly weird (I think there's a hacky way to fix it if anyone complains about that, I could do another PR).

Karen fixed that issue in 1.30 https://github.com/streamlit/streamlit/pull/7921 in a way that totally broke this prometheus collector. So the only version that won't work at all is 1.30. Then he fixed it to be compatible with this AND have the bug fix in 1.31.

I think it's OK with the note but LMK if you think we should handle another way!

Reply:
Thanks a lot for those precisions. I was mostly being careful for ImportErrors, but 1.12 is already a while back! 
 Let's do it!